### PR TITLE
Add a module for easier handling of entity spawning

### PR DESCRIPTION
**Pull request by liach**
replaces #384

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Review by MariaTheDinkus — APPROVED** [review]
Looks much better than my PR, and handles things in what I would consider an ideal way.

File: fabric-networking-entity-v1/src/main/java/net/fabricmc/fabric/api/networking/entity/v1/EntityNetworking.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.network.Packet;
+
+import net.fabricmc.fabric.impl.networking.entity.v1.FabricEntityNetworking;
+
+/**
+ * Fabric entity networking hooks.
+ */
+public interface EntityNetworking {
```
Comment (Pyrofab) [inline fabric-networking-entity-v1/src/main/java/net/fabricmc/fabric/api/networking/entity/v1/EntityNetworking.java:28]: `EntityNetworking` doesn't sound like a class that has instances. I'd either rename it to `EntityNetworkingHelper` like [ResourceManagerHelper](https://github.com/FabricMC/fabric/blob/1.14/fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/api/resource/ResourceManagerHelper.java), or replace the implementation with delegated static methods like [OverworldBiomes](https://github.com/FabricMC/fabric/blob/1.14/fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/OverworldBiomes.java)

Comment (liach) [reply fabric-networking-entity-v1/src/main/java/net/fabricmc/fabric/api/networking/entity/v1/EntityNetworking.java:28]: `EntitySpawnPacketHelper`?

Comment (LemmaEOF) [reply fabric-networking-entity-v1/src/main/java/net/fabricmc/fabric/api/networking/entity/v1/EntityNetworking.java:28]: `EntitySpawnPacketHelper` sounds good to me!

**Comment by liach** [comment]
I actually want to come back to this after I finish #455... the new networking api is just too much better than the old one, and the current code uses the old networking api.

**Comment by LemmaEOF** [comment]
While we wait for the new networking API, something like [this mixin](https://github.com/Boundarybreaker/Aquarius-Fabric/blob/master/src/main/java/space/bbkr/aquarius/mixins/MixinClientPlayNetworkHandler.java#L25-L42) that I currently use for spawning entities in Aquarius might be good. It'd need to be API-ified for Fabric of course, probably with a core construct of a `Map<EntityType, EntitySpawner>` (with `EntitySpawner` being a functional interface that takes the packet, x, y, and z and returns an Entity) that's registered to in a client init.

**Comment by liach** [comment]
After another view, I think I should handle this similar to how juuz handles screen handler v1, where he attaches extra buffer for mods to read on their own. I should consider redoing a new module following suits of #584

**Comment by liach** [comment]
This module appears to obese to me. I will try look into it later and find a more elegant solution

**Pull request closed without merging**